test/batch_node.jl

```diff
@@ -0,0 +1,129 @@
+@testset "AWSBatchNodeManager" begin
+    @testset "AWS_BATCH_JOB_ID DNE" begin
+        withenv(
+            "AWS_BATCH_JOB_MAIN_NODE_INDEX" => 1
+        ) do
+            @test_throws ErrorException AWSBatchNodeManager()
+        end
+    end
+
+    @testset "AWS_BATCH_JOB_MAIN_NODE_INDEX DNE" begin
+        withenv(
+            "AWS_BATCH_JOB_ID" => "job_id"
+        ) do
+            @test_throws ErrorException AWSBatchNodeManager()
+        end
+    end
+
+    @testset "AWS_BATCH_JOB_NUM_NODES DNE" begin
+        withenv(
+            "AWS_BATCH_JOB_ID" => "job_id",
+            "AWS_BATCH_JOB_NODE_INDEX" => 1,
+            "AWS_BATCH_JOB_MAIN_NODE_INDEX" => 1,
+        ) do
+            @test_throws KeyError AWSBatchNodeManager()
+        end
+    end
+
+    @testset "AWS_BATCH_JOB_NODE_INDEX != AWS_BATCH_JOB_MAIN_NODE_INDEX" begin
+        withenv(
+            "AWS_BATCH_JOB_ID" => "job_id",
+            "AWS_BATCH_JOB_NODE_INDEX" => 0,
+            "AWS_BATCH_JOB_MAIN_NODE_INDEX" => 1
+        ) do
+            @test_throws ErrorException AWSBatchNodeManager()
+        end
+    end
+
+    @testset "AWS_BATCH_JOB_NUM_NODES -- String" begin
+        withenv(
+            "AWS_BATCH_JOB_ID" => "job_id",
+            "AWS_BATCH_JOB_MAIN_NODE_INDEX" => 1,
+            "AWS_BATCH_JOB_NODE_INDEX" => 1,
+            "AWS_BATCH_JOB_NUM_NODES" => "foobar"
```

Comment:
Will this pass if it's a "1"?

Reply:
Yes, we attempt to set the `num_workers` by `parse(Int, N)`. I can add in a test for this.